### PR TITLE
[BOLT][RISCV] Carry-over annotations when fixing calls

### PR DESCRIPTION
**Pull request by jobnoorman**
`FixRISCVCallsPass` changes all different forms of calls to `PseudoCALL` instructions. However, the original call's annotations were lost in the process.

This patch fixes this by moving all annotations from the old to the new call. `MCPlusBuilder::moveAnnotations` had to be made public for this.

### CODE REVIEW

File: bolt/lib/Passes/FixRISCVCallsPass.cpp [hunk 1]
```diff
@@ -39,8 +41,19 @@ void FixRISCVCallsPass::runOnFunction(BinaryFunction &BF) {
         auto *Target = MIB->getTargetSymbol(*II);
         assert(Target && "Cannot find call target");
 
+        auto OldCall = *NextII;
```
Comment (yota9) [inline bolt/lib/Passes/FixRISCVCallsPass.cpp:44]: We've got a policy to not using auto if it is not necessary, please minimise their usage

**Review by yota9 — APPROVED** [review]
overall lgtm. I had the same problems with golang, be aware that BF.fixBranches() may replace instructions and not preserve annotations too

File: bolt/lib/Passes/FixRISCVCallsPass.cpp (hunk 1 above)
Comment (yota9) [reply bolt/lib/Passes/FixRISCVCallsPass.cpp:44]: Let the Meta team members to clarify, they even has patches for auto expansion and asked me couple years ago. Maybe the policy changed. At least the LLVM coding standard says:
>> Don’t “almost always” use auto, but do use auto with initializers like cast<Foo>(...) or other places where the type is already obvious from the context. 
E.g. here it is obviously MCInst or MCSymbol above & etc. I don't really insist, but as for me I don't really like autos too except for complex types with smth like returning std::vector with std::pair elements..

Comment (aaupov) [reply bolt/lib/Passes/FixRISCVCallsPass.cpp:44]: Yes, we should stick to LLVM policy of only using auto in small number of cases (the type is obvious (casts), the type doesn't matter (lambdas), template type). Making type explicit helps with reviewing where we have limited context window by default.

**Comment by aaupov** [comment]
Looks good but please expand the autos before merging.

**Pull request merged**